python/cuml/cuml/manifold/spectral_embedding.pyx

```diff
@@ -0,0 +1,268 @@
+        The dimension of the projection subspace.
+
+    random_state : int, RandomState instance or None, default=None
+        A pseudo random number generator used for the initialization.
+        Use an int to make the results deterministic across calls.
+
+    n_neighbors : int or None, default=None
+        Number of nearest neighbors for nearest_neighbors graph building.
+        If None, n_neighbors will be set to max(n_samples/10, 1).
+        Only used when A has shape (n_samples, n_features).
+
+    norm_laplacian : bool, default=True
+        If True, then compute symmetric normalized Laplacian.
+
+    drop_first : bool, default=True
+        Whether to drop the first eigenvector. For spectral embedding, this
+        should be True as the first eigenvector should be constant vector for
+        connected graph, but for spectral clustering, this should be kept as
+        False to retain the first eigenvector.
+
+    Returns
+    -------
+    embedding : cupy.ndarray of shape (n_samples, n_components)
+        The reduced samples.
+
+    Notes
+    -----
+    Spectral Embedding (Laplacian Eigenmaps) is most useful when the graph
+    has one connected component. If there graph has many components, the first
+    few eigenvectors will simply uncover the connected components of the graph.
+
+    Examples
+    --------
+    >>> import cupy as cp
+    >>> from cuml.manifold import spectral_embedding
+    >>> X = cp.random.rand(100, 20, dtype=cp.float32)
+    >>> embedding = spectral_embedding(X, n_components=2, random_state=42)
+    >>> embedding.shape
+    (100, 2)
+    """
+
+    if handle is None:
+        handle = Handle()
+    cdef device_resources *h = <device_resources*><size_t>handle.getHandle()
+
+    A, _n_rows, _n_cols, _ = \
+        input_to_cuml_array(A, order="C", check_dtype=np.float32,
+                            convert_to_dtype=cp.float32)
+    A_ptr = <uintptr_t>A.ptr
+
+    config.n_components = n_components
+    config.seed = check_random_seed(random_state)
+
+    config.n_neighbors = (
+        n_neighbors
+        if n_neighbors is not None
+        else max(int(A.shape[0] / 10), 1)
+    )
+
+    config.norm_laplacian = norm_laplacian
+    config.drop_first = drop_first
+
+    if config.drop_first:
+        config.n_components += 1
+
+    eigenvectors = CumlArray.empty((A.shape[0], n_components), dtype=A.dtype, order='F')
+
+    eigenvectors_ptr = <uintptr_t>eigenvectors.ptr
+
+    spectral_embedding_cuvs(
+        deref(h), config,
+        make_device_matrix_view[float, int, row_major](
+            <float *>A_ptr, <int> A.shape[0], <int> A.shape[1]),
+        make_device_matrix_view[float, int, col_major](
+            <float *>eigenvectors_ptr, <int> A.shape[0], <int> n_components))
+
+    return eigenvectors
+
+
+class SpectralEmbedding(Base,
+                        CMajorInputTagMixin,
+                        SparseInputTagMixin):
+    """Spectral embedding for non-linear dimensionality reduction.
```

Comment:
Nice docstring!

Can you add a section to the API docs so it's rendered? Should go under this heading here (https://github.com/rapidsai/cuml/blob/992347da5dc49b66cbb2a05bc59a343d970ecc47/docs/source/api.rst?plain=1#L430), with a new subheading (`Spectral Embedding`?):